syntax = "proto2";

package oppie.comm;

import "google/protobuf/timestamp.proto";
import "google/protobuf/any.proto";

// Common step structure used in TaskCard
message Step {
  required string instruction = 1;
  optional string expected_outcome = 2;
}

// Common artifact structure used in ResultCard
message Artifact {
  required string id = 1;
  required string type = 2; // Such as "file", "data_blob", etc.
  required string location = 3; // File path or reference
  optional bytes content = 4; // If carrying content directly
  optional string content_type = 5; // MIME type
}

// Log entry for diagnostics
message LogEntry {
  required string level = 1; // INFO, WARNING, ERROR, etc.
  required string message = 2;
  required google.protobuf.Timestamp timestamp = 3;
  optional string source = 4; // Log source module/component
  map<string, string> context = 5; // Context key-value pairs
}

// OpenTelemetry trace context for cross-protocol request tracing
message TraceContext {
  required string trace_id = 1;
  required string span_id = 2;
  optional bool sampled = 3;
  map<string, string> baggage = 4; // Additional context data
} 